components/dashboard/src/service/json-rpc-scm-client.ts

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { PartialMessage } from "@bufbuild/protobuf";
+import { PromiseClient } from "@connectrpc/connect";
+import { SCMService } from "@gitpod/public-api/lib/gitpod/v1/scm_connect";
+import { converter } from "./public-api";
+import { getGitpodService } from "./service";
+import { ApplicationError, ErrorCodes } from "@gitpod/gitpod-protocol/lib/messaging/error";
+import {
+    GetSCMTokenRequest,
+    GuessTokenScopesRequest,
+    SearchRepositoriesRequest,
+    ListSuggestedRepositoriesRequest,
+    GetSCMTokenResponse,
+    ListSuggestedRepositoriesResponse,
+    SearchRepositoriesResponse,
+    GuessTokenScopesResponse,
+} from "@gitpod/public-api/lib/gitpod/v1/scm_pb";
+
+export class JsonRpcScmClient implements PromiseClient<typeof SCMService> {
+    async getSCMToken({ host }: PartialMessage<GetSCMTokenRequest>): Promise<GetSCMTokenResponse> {
+        if (!host) {
+            throw new ApplicationError(ErrorCodes.BAD_REQUEST, "host is required");
+        }
+        const response = await getGitpodService().server.getToken({ host });
+        return new GetSCMTokenResponse({
+            token: response ? converter.toSCMToken(response) : undefined,
+        });
+    }
+
+    async guessTokenScopes({
+        gitCommand,
+        host,
+        repoUrl,
+    }: PartialMessage<GuessTokenScopesRequest>): Promise<GuessTokenScopesResponse> {
+        if (!host) {
+            throw new ApplicationError(ErrorCodes.BAD_REQUEST, "host is required");
+        }
+        const response = await getGitpodService().server.guessGitTokenScopes({
+            gitCommand: gitCommand || "",
+            host,
+            repoUrl: repoUrl || "",
+        });
+        return new GuessTokenScopesResponse({
+            message: response.message,
```

Comment:
Same there NotFound or make it something like default empty array? (not sure if new GuessTokenScopesResponse will give it default value or not)